Added to gas in binutils commit 685080f2100373d1a45932521f9a24922a68d68f

--- origsrc/gcc-5.2.0/gcc/config/v850/t-v850	2015-01-05 06:33:28.000000000 -0600
+++ src/gcc-5.2.0/gcc/config/v850/t-v850	2015-07-28 19:03:45.472109200 -0500
@@ -17,8 +17,8 @@
 # <http://www.gnu.org/licenses/>.
 
 # Create target-specific versions of the libraries
-MULTILIB_OPTIONS  = m8byte-align mgcc-abi msoft-float
-MULTILIB_DIRNAMES =  8byte        gcc-abi  soft-float
+MULTILIB_OPTIONS  = m8byte-align mgcc-abi
+MULTILIB_DIRNAMES =  8byte        gcc-abi
 
 MULTILIB_OPTIONS  += mv850e3v5
 MULTILIB_DIRNAMES +=  v850e3v5
