https://gcc.gnu.org/bugzilla/show_bug.cgi?id=67061

--- origsrc/gcc-5.2.0/gcc/config/sh/sh-protos.h	2015-02-26 13:14:23.000000000 -0600
+++ src/gcc-5.2.0/gcc/config/sh/sh-protos.h	2015-07-30 00:22:00.519281100 -0500
@@ -198,7 +198,8 @@
     {
       if (BARRIER_P (result.insn))
 	break;
-      if (!NONJUMP_INSN_P (result.insn))
+      if (!NONJUMP_INSN_P (result.insn)
+         && !(REGNO (reg) < FIRST_PSEUDO_REGISTER && CALL_P (result.insn)))
 	continue;
       if (reg_set_p (reg, result.insn))
 	{
