https://gcc.gnu.org/ml/gcc-patches/2015-07/msg02677.html

--- cross-gcc-5.2.0-1.x86_64/origsrc/gcc-5.2.0/gcc/config/m32r/m32r.c	2015-01-15 07:28:42.000000000 -0600
+++ cross-gcc-5.2.0-1.x86_64/src/gcc-5.2.0/gcc/config/m32r/m32r.c	2015-07-31 11:03:09.299389700 -0500
@@ -129,6 +129,7 @@ static bool m32r_can_eliminate (const in
 static void m32r_conditional_register_usage (void);
 static void m32r_trampoline_init (rtx, tree, rtx);
 static bool m32r_legitimate_constant_p (machine_mode, rtx);
+static bool m32r_attribute_identifier (const_tree);
 
 /* M32R specific attributes.  */
 
@@ -145,6 +146,8 @@ static const struct attribute_spec m32r_
 /* Initialize the GCC target structure.  */
 #undef  TARGET_ATTRIBUTE_TABLE
 #define TARGET_ATTRIBUTE_TABLE m32r_attribute_table
+#undef  TARGET_ATTRIBUTE_TAKES_IDENTIFIER_P
+#define TARGET_ATTRIBUTE_TAKES_IDENTIFIER_P m32r_attribute_identifier
 
 #undef TARGET_LEGITIMATE_ADDRESS_P
 #define TARGET_LEGITIMATE_ADDRESS_P m32r_legitimate_address_p
@@ -420,6 +423,14 @@ m32r_handle_model_attribute (tree *node
 
   return NULL_TREE;
 }
+
+static bool
+m32r_attribute_identifier (const_tree name)
+{
+  return strcmp (IDENTIFIER_POINTER (name), "model") == 0
+    ||   strcmp (IDENTIFIER_POINTER (name), "__model__") == 0;
+}
+
 
 /* Encode section information of DECL, which is either a VAR_DECL,
    FUNCTION_DECL, STRING_CST, CONSTRUCTOR, or ???.
